File: Data Structures & Algorithms/combination-target-sum-ii/submission-0.cpp
class Solution {
public:
    vector<vector<int>> ans;

    void backtrack(vector<int>& nums , int target ,vector<int>&d  , int i){
        if(target < 0 || i > nums.size()){
            return;
        }
        
        if(target == 0){
            ans.push_back(d);
            return;
        }

        int temp = nums[i];
        d.push_back(temp);
        backtrack(nums , target - temp  , d , i+1);
        d.pop_back();
        while(temp == nums[i]){
            i+=1;
        }
        backtrack(nums , target  , d , i);
    }
    vector<vector<int>> combinationSum2(vector<int>& candidates, int target) {
        sort(candidates.begin() , candidates.end());
        vector<int>d;
        backtrack(candidates , target , d , 0);
        return ans;
    }
};

File: Data Structures & Algorithms/move-zeroes/submission-0.java
class Solution {
    public void moveZeroes(int[] nums) {
        for(int i =0 ; i<nums.length;i++){
            if(nums[i] != 0){
                int j = i;
                while(j-1 >= 0 && nums[j-1] == 0){
                    nums[j-1] = nums[j];
                    nums[j] = 0;
                    j-=1;
                }
            }
        }
    }
}